httpx/client.py

```diff
@@ -582,15 +582,20 @@ def redirect_stream(
     async def send_handling_auth(
         self,
         request: Request,
+        history: typing.List[Response],
         auth: Auth,
         timeout: Timeout,
         verify: VerifyTypes = None,
         cert: CertTypes = None,
     ) -> Response:
         auth_flow = auth(request)
         request = next(auth_flow)
+        counter = -1
```

Comment:
I understand the intent, but I'd agree it's not immediately obvious. One way to make things clearer could be to use a verbosely-named boolean flag:

```python
made_more_than_one_request = False

while True:
    response = ...

    if made_more_than_one_request:
        history.append(response)

    try:
        # ...
    else:
        request = next_request
        made_more_than_one_request = True
```

Reply:
Fixed. Yeah, this is better. Thanks.